File: Scraper/Store.Scraper/Services/ScraperService.cs
using Microsoft.Extensions.Logging;
using Store.Scraper.Data;

namespace Store.Scraper.Services;

public class ScraperService
{
    protected readonly IScraperRepository Repository;
    protected readonly ILogger<ScraperService> Logger;

    public ScraperService(IScraperRepository repository, ILogger<ScraperService> logger)
    {
        Repository = repository;
        Logger = logger;
    }

    /// <summary>
    /// Collects products and persists them to the database.
    /// </summary>
    /// <remarks>
    /// Use Repository.UpsertProductAsync(product) to save each collected product.
    /// Use Guid.NewGuid() to generate an Id for each product.
    ///
    /// You decide how to collect the data — HttpClient, Playwright, HtmlAgilityPack, etc.
    /// </remarks>
    public virtual async Task ScrapeAsync()
    {
        // TODO: Implement the scraping logic here.
        throw new NotImplementedException("Implement the scraping logic here.");
    }
}

File: Scraper/store-scraper/src/scraperService.ts
import { ScraperRepository } from './repository';

export class ScraperService {
  protected readonly repository: ScraperRepository;

  constructor(repository: ScraperRepository) {
    this.repository = repository;
  }

  /**
   * Collects products and persists them to the database.
   *
   * Use this.repository.upsertProduct(product) to save each collected product.
   * Use crypto.randomUUID() to generate a GUID for each product's id field.
   *
   * You decide how to collect the data — REST API, Playwright, etc.
   */
  async scrape(): Promise<void> {
    // TODO: Implement the scraping logic here.
    throw new Error('Not implemented: implement the scrape() method.');
  }
}
